fc/fc_app.py

```diff
@@ -219,124 +231,162 @@ def estimate_job_size(num_tasks):
-           year,
-           tag):
-
+@ui.config_option
+@ui.verbose_option
+@ui.pass_index(app_name=APP_NAME)
+def submit(index: Index,
+           app_config: str,
+           project: str,
+           queue: str,
+           no_qsub: bool,
+           time_range: Tuple[datetime, datetime],
+           tag: str):
     _LOG.info('Tag: %s', tag)
 
-    qsub = QSubLauncher(norm_qsub_params(
-        {'project': project,
-         'queue': queue,
-         'name': 'fc-generate-{}'.format(tag),
-         'mem': '4G',
-         'noask': True,
-         'wd': True,
-         'ncpus': 1,
-         'walltime': '1h',
-         'extra_qsub_args': ['-W', 'umask=027']}))
-
-    app_config = Path(app_config).absolute()
-    output_tasks_file = Path(output_tasks_file).absolute()
+    app_config_path = Path(app_config).resolve()
+    app_config = paths.read_document(app_config_path)
+
+    task_desc, task_path = init_task_app(
+        job_type="fc",
+        source_types=[app_config['source_type']],
+        output_types=[app_config['output_type']],
+        # TODO: Use @datacube.ui.click.parsed_search_expressions to allow params other than time from the cli?
+        datacube_query_args=Query(index=index, time=time_range).search_terms,
+        app_config_path=app_config_path,
+        pbs_project=project,
+        pbs_queue=queue
+    )
+    _LOG.info("Created task description: %s", task_path)
 
-    ret_code, qsub_stdout = qsub('generate',
-                                 '-v', '-v',
-                                 '--save-tasks', str(output_tasks_file),
-                                 '--project', project,
-                                 '--queue', queue,
-                                 '--app-config', str(app_config),
-                                 '--year', year,
-                                 '--tag', tag)
+    if no_qsub:
+        _LOG.info('Skipping submission due to --no-qsub')
+        return 0
 
-    _LOG.info('Launched qsub: %d -> %s', ret_code, qsub_stdout)
+    submit_subjob(
+        name='generate',
+        task_desc=task_desc,
+        command=[
+            'generate', '-v', '-v',
+            '--task-desc', str(task_path),
+            '--tag', tag
+        ],
+        qsub_params=dict(
+            mem='4G',
+            wd=True,
+            ncpus=1,
+            walltime='1h',
+            name='fc-generate-{}'.format(tag)
+        )
+    )
 
 
 @cli.command(help='Generate Tasks into file and Queue PBS job to process them')
-@click.option('--project', '-P', default='u46')
-@click.option('--queue', '-q', default='normal',
-              type=click.Choice(['normal', 'express']))
-@click.option('--year', 'time_range', callback=task_app.validate_year, help='Limit the process to a particular year')
-@click.option('--no-qsub', 'no_qsub', is_flag=True, default=False, help="Skip submitting qsub for next step")
+@click.option('--no-qsub', is_flag=True, default=False, help="Skip submitting qsub for next step")
+@click.option(
```

Comment:
We could even get really crazy and make a new decorator for this option, that does the whole `_make_config_and_description()` thing. But I'm quite ambivalent about the idea